pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -123,41 +145,44 @@ def get_latest_available_date(
             f"Getting latest available date for the table: {table_name} "
             f"until {until}"
         )
-        table = get_metastore_table(table_name, self.tables)
-
-        def before_until(date: datetime.date) -> bool:
-            return date <= until  # type: ignore
+        metastore_table = get_metastore_table(table_name, self.tables)
 
-        def extract_dates(file_name: str) -> datetime.date:
-            return convert_str_to_date(
-                file_name.split("=")[1],
-                fmt=table.info_date_settings.format,
+        if metastore_table.format == TableFormat.delta:
+            df_select = self._read_table(
+                metastore_table.format, metastore_table.path
+            ).select(f"{metastore_table.info_date_settings.column}")
+            dates_list = [data[0] for data in df_select.distinct().collect()]
```

Comment:
Take a look here, not sure what are you doing, but I am confused by the .collect and not readable data[0] for data.

Reply:
`collect()` returns records as an array, and `[0]` is the first column. At least this is how it works in Scala.
How would you do it differently?

Reply:
something like this would be more sparktonic based on my sense of the spark nature:):
```python
latest_date =df.select("info_date").distinct().rdd.map(lambda x: x[0]).max()
```

Reply:
From tastes perspective I like the original code. Seen many Spark functions that do this trick:
Quoting my own code here, but have seen similar things in other code bases, and it does not trigger a wtf for me 😄  : https://github.com/AbsaOSS/pramen/blob/main/pramen/core/src/main/scala/za/co/absa/pramen/core/reader/TableReaderJdbc.scala#L79-L79

Extracting the processing logic as a lambda won't do much since you can't avoid `data[0]` or something like this.
Do I understand it correctly that you propose:
```python
def get_info_date_value(row: Row) -> str:
    return str[0]
```

Also, I'm not sure why you want to convert to an rdd - seems like redundant to me, might be even less performant.

Reply:
I'd be happy to chat about this tomorrow, sure do not block on this!

Reply:
Sure, thanks!

Reply:
Maybe the low-hanging fruit here would be to rename `data` to `row` which (maybe) makes it clearer that we are extracting the first column but I guess it's still pretty much a matter of preference. I have used this "pattern" many times so to me it was familiar.